src/restLevelClient/generateRestLevel.ts

```diff
@@ -10,23 +10,34 @@ import { generateSchemaTypes } from "./generateSchemaTypes";
 import { format } from "prettier";
 import { prettierJSONOptions, prettierTypeScriptOptions } from "./config";
 import { generateParameterInterfaces } from "./generateParameterTypes";
-import { generatePathFirstClient } from "./generateClient";
+import { generatePathFirstClient } from "./generateClientDefinition";
+import { generateClient } from './generateClient';
 import { generateIndexFile } from "../generators/indexGenerator";
 import { generatePagingHelper } from "./generatePagingHelper";
 import { generatePollingHelper } from "./generatePollingHelper";
+import { generateTopLevelIndexFile } from './generateTopLevelIndexFile';
 import { hasPagingOperations } from "../utils/extractPaginationDetails";
 import { hasPollingOperations } from "./helpers/hasPollingOperations";
+import { NameType, normalizeName } from "../utils/nameUtils";
 import { generateKarmaConfigFile } from "../generators/static/karmaConfigFileGenerator";
 import { generateEnvFile } from "../generators/test/envFileGenerator";
 import { generateEnvBrowserFile } from "../generators/test/envBrowserFileGenerator";
 import { generateRecordedClientFile } from "../generators/test/recordedClientFileGenerator";
 import { generateSampleTestFile } from "../generators/test/sampleTestGenerator";
+
+const batchOutputFolder: [string, string, string][] = [];
 /**
  * Generates a Rest Level Client library
  */
 export async function generateRestLevelClient() {
   const host = getHost();
   const { model } = getSession();
+  const { batch, srcPath } = getAutorestOptions();
+  if (srcPath) {
```

Comment:
it feels like this should be inside your if block in line 76? 

Would it be possible to move this check and the check in line 76 inside the `generateTopLevelIndexFile` function? If the conditions are not met `generateTopLevelIndexFile` does nothing. This will keep this function cleaner